homeassistant/components/vodafone_station/config_flow.py

```diff
@@ -0,0 +1,138 @@
+"""Config flow for Vodafone Station integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from aiovodafone import VodafoneStationApi, exceptions as aiovodafone_exceptions
+import voluptuous as vol
+
+from homeassistant import core, exceptions
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import _LOGGER, DEFAULT_HOST, DEFAULT_USERNAME, DOMAIN
+
+
+def user_form_schema(user_input: dict[str, Any] | None) -> vol.Schema:
+    """Return user form schema."""
+    user_input = user_input or {}
+    return vol.Schema(
+        {
+            vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+            vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+        }
+    )
+
+
+STEP_REAUTH_DATA_SCHEMA = vol.Schema({vol.Required(CONF_PASSWORD): str})
+
+
+async def validate_input(
+    hass: core.HomeAssistant, data: dict[str, Any]
+) -> dict[str, str]:
+    """Validate the user input allows us to connect."""
+
+    api = VodafoneStationApi(data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD])
+
+    try:
+        await api.login()
+    except aiovodafone_exceptions.CannotConnect as err:
+        raise CannotConnect from err
+    except aiovodafone_exceptions.CannotAuthenticate as err:
+        raise InvalidAuth from err
```

Comment:
You can raise these directly and capture them below. There is no need to re-wrap the exceptions

Reply:
Done